ser4ph22/weatherwise
# feat: initial project setup

## Initial Project Setup

### Description
This PR sets up the initial project structure for WeatherWise, including development environment configuration, project structure, and CI/CD workflows.

### Changes Made
- Initialize Git repository with branch structure
- Configure VS Code settings and recommended extensions
- Create basic project directory structure
- Set up GitHub workflows for CI/CD
- Initialize NPM project with essential dependencies

### Type of Change

### Development Setup Completed
- [x] VS Code configuration
  - Settings for formatting and linting
  - Recommended extensions list
  - Created all required directories
  - Set up initial configuration files
- [x] GitHub Workflows
  - CI pipeline for testing and building
  - Deployment workflow template
  - Core React dependencies
  - Development tools and TypeScript
  - Build and styling utilities

### Test Plan
1. Clone repository and check out develop branch
2. Install recommended VS Code extensions
3. Run `pnpm install`
4. Verify project structure matches documentation
5. Confirm GitHub workflows are properly configured

### Checklist
- [x] Project structure matches documentation
- [x] GitHub workflows are properly configured
- [x] All configuration files are present
- [x] Dependencies are properly specified

### Additional Notes
This is the initial setup PR. After merging, we can proceed with:
1. Setting up the basic React application
2. Configuring Tailwind CSS
3. Setting up WeatherAPI integration

### Related Issues
None
